mapwms.c

```diff
@@ -376,7 +372,7 @@ int msWMSApplyFilter(mapObj *map, int version, const char *filter,
       continue;
 
     /* Skip empty filters */
-    if (paszFilters[curfilter][0] == '\0') {
+    if (paszFilters[ows_request->layerwmsfilterindex[lp->index]][0] == '\0') {
```

Comment:
Use curfilter = ows_request->layerwmsfilterindex[lp->index] and reuse it aftward